components/DefaultScreen.js

```diff
@@ -1,16 +1,99 @@
-import React from 'react';
-import { View, ImageBackground, StyleSheet, Text, StatusBar } from 'react-native';
+import React, {useEffect,useContext,useState} from 'react';
+import { View, StyleSheet, Text, StatusBar } from 'react-native';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
 import Footer from './Footer';
+import * as Location from 'expo-location';
+import { AuthContext } from '../navigation';
+
 
 const DefaultScreen = () => {
+   
+  const {isLocationEnabled, setIsLocationEnabled, hasPubcrawl, setHasPubcrawl} = useContext(AuthContext);
+  const [isVisible, setIsVisible] = useState(false);
+  
+  // Show the text after 500ms (avoid seeing it if you don't stay on the screen for long)
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setIsVisible(true);
+    }, 500);
+    
+    // Cleanup the timer when the component unmounts or the state changes
+    return () => clearTimeout(timer);
+  }, []);
+
+  //check location permission
+  const checkLocationPermission = async () => {
+    try {
+      const { status } = await Location.requestForegroundPermissionsAsync();
+      if (status === 'granted') {
+        console.log('Location permission is granted');
+        if (!isLocationEnabled) {
+        setIsLocationEnabled(true);
+        }
+      } else {
+      }
+    } catch (error) {
+      console.log('Error checking location permission:', error);
+    }
+  };
+    
+    //check location with a timer of 5 seconds
+    const checkLocation = () => {
+      if (!isLocationEnabled) {
+      setTimeout(() => {
```

Comment:
why not use setInterval instead of use setTimeout with recursive method ?

Reply:
that's what I used before but it caused issues